File: Problems/Algorithms/199. Binary Tree Right Side View/binary_tree_side_view.rs
// Definition for a binary tree node.
// #[derive(Debug, PartialEq, Eq)]
// pub struct TreeNode {
//   pub val: i32,
//   pub left: Option<Rc<RefCell<TreeNode>>>,
//   pub right: Option<Rc<RefCell<TreeNode>>>,
// }
// 
// impl TreeNode {
//   #[inline]
//   pub fn new(val: i32) -> Self {
//     TreeNode {
//       val,
//       left: None,
//       right: None
//     }
//   }
// }
use std::rc::Rc;
use std::cell::RefCell;
use std::collections::VecDeque;

impl Solution {
    pub fn right_side_view(root: Option<Rc<RefCell<TreeNode>>>) -> Vec<i32> {
        let root = match root {
            Some(node) => node,
            None => return vec![],
        };
        
        let mut results = vec![];
        let mut queue = VecDeque::new();
        queue.push_back(root);
        while queue.len() > 0 {
            let size = queue.len();
            for i in 0..size {
                let node = queue.pop_front().unwrap();
                
                if let Some(ref left) = node.borrow().left {
                    queue.push_back(left.clone());
                }
                
                if let Some(ref right) = node.borrow().right {
                    queue.push_back(right.clone());
                }
                
                if i == size-1 {
                    results.push(node.borrow().val);
                }
            }
        }
        
        results
    }
}

File: Problems/Algorithms/199. Binary Tree Right Side View/binary_tree_side_view.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def rightSideView(self, root: Optional[TreeNode]) -> List[int]:
        if root is None:
            return []

        results = []
        queue = [root]
        while len(queue) > 0:
            size = len(queue)
            for i in range(size):
                curr = queue.pop(0)

                if curr.left is not None:
                    queue.append(curr.left)
                if curr.right is not None:
                    queue.append(curr.right)

                if i == size-1:
                    results.append(curr.val)

        return results

